src/include/miopen/any_solver.hpp

```diff
@@ -166,12 +235,46 @@ struct AnySolver
         bool IsTunable() const override { return TunableSolver::Is; }
         bool IsDynamic() const override { return value.IsDynamic(); }
         float GetWti(const ConvolutionContext& ctx) const override { return value.GetWti(ctx); }
+
         ConvSolution FindSolution(const ConvolutionContext& ctx,
                                   Db& db,
                                   const miopen::AnyInvokeParams& invoke_ctx) const override
         {
             return miopen::solver::FindSolution(value, ctx, db, invoke_ctx);
         };
+
+        std::string GetPerfCfgParams(const ConvolutionContext& ctx, Db& db, std::true_type) const
```

Comment:
The body of this function looks like a modified copy of very similar code from `FindSolutionImpl` (src/include/miopen/find_solution.hpp). Can we generalize this?

Reply:
The objective of this function is to return the parameter string from the performance config. I see the portion that is common here, and most likely it was the example for this addition. They are entirely similar in that they 
1. use the same type declaration
2. load from the db
3. check validity of the config loaded
4. do a function x with the loaded config
5. log progress
The main difference being #4 the way to minimize this code would be to create an additional function providing a callback in place of #4.
Disregarding log statements and syntactical spacing there are a total of 4 shared lines. Not sure if it's worth complicating with a callback.

Reply:
If we remove this function, I'll need a way to retrieve the serialized performance parameters. Could be the winner from the solver or from a solution. I'm searching for another method, but nothing yet.

Reply:
Can we add this function to a PR that will actually use it? This will allow us to expedite the review of this one.

Reply:
I see. If you know SolverId, then it is possible to construct an instance of the Solver and then follow the design pattern of:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/72d89b40ee97666efaac0a0979a48680bd02f1da/src/include/miopen/find_solution.hpp#L123-L125
I can do this for you in a future PR where you will implement Fin changes. Or you can ask @DrizztDoUrden to assist.

But let's avoid adding unused code in this PR. Is it acceptable for you?

Reply:
Please send me proposed Fin changes (where you call `GetPerfCfgParams`) and I will offer a substitute. This is necessary, because

> from the architectural point of view, getting tuning config from the database is not a Solver's job.

Reply:
I'll email the code section to your amd address. Do you still have access?

Reply:
@cderb No, unfortunately. But you can send me mail to DXC (please check your amd inbox).